oop/src/PurchaseOrders/infrastructure/repositories/PurchaseOrder.repo.ts

```diff
@@ -1,15 +1,30 @@
 import { UUID } from "../../../utilities/uuid";
 import { IPORepository } from "../../domain/PurchaseOrder.repo-interface";
 import { PurchaseOrder } from "../../domain/PurchaseOrder.entity";
+import { PONumber } from "../../domain/PONumber.value-object";
+
+const byPONumber = (a: PurchaseOrder, b: PurchaseOrder) => {
+  return (a?.number?.digits ?? 0) > (b?.number?.digits ?? 0) ? -1 : 1;
+};
 
 export class PORepository implements IPORepository {
   purchaseOrders: PurchaseOrder[] = [];
   async save(po: PurchaseOrder) {
-    this.purchaseOrders.push(po);
+    const existingPO = this.purchaseOrders.find((p) => p.id === po.id);
+    this.purchaseOrders = existingPO
+      ? [...this.purchaseOrders.filter((p) => p.id !== po.id), po]
+      : [...this.purchaseOrders, po];
     return undefined;
   }
   async fetch(id: UUID) {
     const po = this.purchaseOrders.find((p) => p.id === id);
     return po || null;
   }
+  async fetchLastPOByNumber() {
```

Comment:
I originally had this as a `nextPONumber` function but decided at the last minute to change it to get the last po. That _seemed_ like it might be more general purpose and it keeps some of the logic out of the repo.